Resources/doc/reference/security.rst

```diff
@@ -64,99 +64,101 @@ The Strategy class must implement the ``DownloadStrategyInterface`` which contai
-    use Symfony\Component\Security\Core\SecurityContextInterface;
+    use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
     use Symfony\Component\Translation\TranslatorInterface;
-    use Symfony\Component\DependencyInjection\ContainerInterface;
 
-    class SessionDownloadStrategy implements DownloadStrategyInterface
+    class UsersDownloadStrategy implements DownloadStrategyInterface
     {
-        protected $container;
+        /**
+         * @var TokenStorageInterface
+         */
+        protected $tokenStorage;
 
+        /**
+         * @var TranslatorInterface
+         */
         protected $translator;
 
-        protected $times;
-
-        protected $sessionKey = 'sonata/media/session/times';
+        /**
+         * @var array
+         */
+        protected $users;
 
         /**
-         * @param \Symfony\Component\Translation\TranslatorInterface $translator
-         * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
-         * @param int $times
+         * @param TokenStorageInterface $tokenStorage
+         * @param TranslatorInterface   $translator
+         * @param array                 $users
          */
-        public function __construct(TranslatorInterface $translator, ContainerInterface $container, $times)
+        public function __construct(TokenStorageInterface $tokenStorage, TranslatorInterface $translator, array $users = array())
         {
-            $this->times    = $times;
-            $this->container = $container;
+            $this->tokenStorage = $tokenStorage;
             $this->translator = $translator;
+            $this->users = $users;
         }
 
         /**
-         * @param \Sonata\MediaBundle\Model\MediaInterface $media
-         * @param \Symfony\Component\HttpFoundation\Request $request
-         * @return bool
+         * {@inheritdoc}
          */
         public function isGranted(MediaInterface $media, Request $request)
         {
-            if (!$this->container->has('session')) {
-                return false;
-            }
-
-            $times = $this->getSession()->get($this->sessionKey, 0);
-
-            if ($times >= $this->times) {
-                return false;
-            }
-
-            $this->getSession()->set($this->sessionKey, $times + 1);
-
-            return true;
+            return in_array($this->tokenStorage->getToken()->getUsername(), $this->users);
         }
 
         /**
-         * @return string
+         * {@inheritdoc}
          */
         public function getDescription()
         {
-            return $this->translator->trans('description.session_download_strategy', array('%times%' => $this->times), 'SonataMediaBundle');
-        }
-
-        /**
-         * @return \Symfony\Component\HttpFoundation\Session
-         */
-        private function getSession()
-        {
-            return $this->container->get('session');
+            return $this->translator->trans(
```

Comment:
Oh right, I thought you had forgotten to call `transChoice` for the other call because you seem confused about it. Nevermind this comment.